### PR TITLE
add FreeBSD support

### PR DESCRIPTION
**Pull request by olevole**

### CODE REVIEW

File: spec/classes/rabbitmq_spec.rb [hunk 1]
```diff
@@ -1384,6 +1384,17 @@
     end
   end
 
+  context "on FreeBSD" do
+    with_freebsd_facts
+    it 'installs the rabbitmq package' do
+      should contain_package('rabbitmq-server').with(
+        'ensure'   => 'installed',
+        'name'     => 'rabbitmq',
+        'provider' => 'freebsd'
```
Comment (wyardley) [inline spec/classes/rabbitmq_spec.rb]: This is why your test is failing, it's expecting pkgng.
Though I think provider isn't actually needed. Not sure if #493 will get merged, but would have to adjust slightly to get FreeBSD support working if / when that gets merged.

File: manifests/params.pp [hunk 2]
```diff
@@ -38,6 +38,17 @@
       $rabbitmq_home    = '/var/rabbitmq'
       $plugin_dir       = '/usr/local/lib/rabbitmq/plugins'
     }
+    'FreeBSD': {
+      $package_ensure   = 'installed'
+      $package_name     = 'rabbitmq'
+      $service_name     = 'rabbitmq'
+      $package_provider = 'pkgng'
+      $version          = '3.4.2'
```
Comment (wyardley) [inline manifests/params.pp]: Also, keep in mind that while `$version` is used internally as things work now, it's ignored when using the system package provider, so it doesn't really matter in this case.

**Comment by wyardley** [comment]
ps - https://github.com/voxpupuli/puppet-rabbitmq/blob/master/metadata.json may also need to be updated (FreeBSD and OpenBSD support are both not currently included there)

**Comment by wyardley** [comment]
@olevole would you like to rework this one?

**Comment by olevole** [comment]
@wyardley Hi Will! 
Thank you very much for being able to spare time and for comments. I took into account your wishes. Unfortunately, I now have a very gigantic problem with time at the moment, sooner or later I will be able to finish and re-send patch. If you can quickly fix my shortcomings - don't wait for me, I'll be only too glad to do it and ready to test module on my FreeBSD cluster's. I apologize for the delay.

**Comment by wyardley** [comment]
I can take a stab at it once #493 is merged, shouldn't be hard.

**Comment by wyardley** [comment]
@olevole ps - one question, though - is `pkgng` the default provider in Puppet on FreeBSD? I'm pretty familiar with FreeBSD (a user for years), but haven't run Puppet with it before.

**Comment by wyardley** [comment]
Haven't submitted yet, but check out this:
https://github.com/voxpupuli/puppet-rabbitmq/compare/master...wyardley:freebsd?expand=1
Will update it after we merge #493.

**Comment by wyardley** [comment]
reworking via #567

**Pull request closed without merging**